### PR TITLE
Add three new buttons

### PR DESCRIPTION
**Pull request by xatza**
Hello,
Specifically i add x^y button,which calculates x(first number) power of y(second number).Second button is log10(x),which calculates the logarithm of a number x.Finally i add x% button,which calculates the percentage of the number x.
I hope you find it helpful.

Friendly,
Donis Chantzaridis

2f4e7b4079cfb65c4c1d7128b06801ae

### CODE REVIEW

File: src/simplejavacalculator/UI.java [hunk 1]
```diff
@@ -59,6 +59,9 @@ public UI() {
         butCos = new JButton("Cos");
         butSin = new JButton("Sin");
         butTan = new JButton("Tan");
+        butxpowerofy=new JButton("x^y");
```
Comment (pH-7) [inline src/simplejavacalculator/UI.java]: Can you add a blank space between `=`, like `butxpowerofy = new JButton("x^y");`

File: src/simplejavacalculator/UI.java [hunk 2]
```diff
@@ -59,6 +59,9 @@ public UI() {
...
+        butlog=new JButton("log10(x)");
```
Comment (pH-7) [inline src/simplejavacalculator/UI.java]: Same here

File: src/simplejavacalculator/UI.java [hunk 3]
```diff
@@ -59,6 +59,9 @@ public UI() {
...
+        butrate=new JButton("x%");
```
Comment (pH-7) [inline src/simplejavacalculator/UI.java]: Same here ;)

File: src/simplejavacalculator/UI.java [hunk 4]
```diff
@@ -31,7 +31,7 @@
     private final JTextArea text;
     private final JButton but[], butAdd, butMinus, butMultiply, butDivide,
             butEqual, butCancel, butSquareRoot, butSquare, butOneDevidedBy,
-            butCos, butSin, butTan;
+            butCos, butSin, butTan,butxpowerofy,butlog,butrate;
```
Comment (pH-7) [inline src/simplejavacalculator/UI.java]: Can you add a space after each comma?

File: src/simplejavacalculator/Calculator.java [hunk 5]
```diff
@@ -78,6 +85,14 @@ public Double calculateMono(MonoOperatorModes newMode, Double num) {
         if (newMode == MonoOperatorModes.tan) {
             return Math.tan(num);
         }
+        if (newMode == MonoOperatorModes.log) {
+            return log10(num);
+        }
+        if (newMode == MonoOperatorModes.rate) {
+           return num/100;
```
Comment (pH-7) [inline src/simplejavacalculator/Calculator.java]: I know, its a small detail, but for consistency and standard, can you add space between the `/`
```java
return num / 100;
```

Thanks!

**Comment by pH-7** [comment]
@xatza Just a few really minor changes ;)
Once done, I will be happy to merge it 😃 

Thanks again!! Really appreciate

**Comment by pH-7** [comment]
Oh, @xatza, can also update `SimpleJavaCalculator.jar` with your new changes (so it will be updated).

Normally you should be able to generate a new one with the following when you are in `src/` folder.

```bash
jar cmvf ../manifest.mf ../SimpleJavaCalculator.jar simplejavacalculator/*.class
```

Then, execute `java -jar SimpleJavaCalculator.jar` to make sure it works well (the calculator should open).

**Comment by xatza** [comment]
Ok @pH-7 i made all compulsory changes,i also update .jar file.Check and let me know if you want something else :)

**Comment by pH-7** [comment]
Just a last point I notice (just now actually), it's that the "0" goes strangely on the border, beside the calculator's screen (probably because of the additional buttons). Should just be a UI/position fix...?

This is what I was getting before 
screenshot

**Comment by xatza** [comment]
I think i solve the position problem and upload the .jar file too.Please check :)

**Comment by pH-7** [comment]
Nice! Not sure for you, but for me, I got the two last buttons corrupted. The "=" and "C". Probably the hight of the frame should be higher

screen shot 2018-01-10 at 18 55 07

**Comment by xatza** [comment]
Oh sorry.I fix it,i update .jar file too.
Check :)

**Comment by pH-7** [comment]
No worries Kydon ;) Looks good now!
Thanks again for contributing!! Highly appreciated! And feel free to come back later for working again together! 

Have a great lovely weekend 👍

**Pull request merged**